nixos/modules/config/nix-channel.nix

```
@@ -49,22 +79,30 @@ in
   config = mkIf cfg.enable {
 
     environment.extraInit =
-      ''
+      mkIf cfg.channel.enable ''
         if [ -e "$HOME/.nix-defexpr/channels" ]; then
           export NIX_PATH="$HOME/.nix-defexpr/channels''${NIX_PATH:+:$NIX_PATH}"
         fi
       '';
 
+    environment.extraSetup = mkIf (! cfg.channel.enable) ''
```

```suggestion
    environment.extraSetup = mkIf (!cfg.channel.enable) ''
```
I don't know, probably a nit